service/src/test/java/org/broadinstitute/listener/relay/UtilTest.java

```diff
@@ -49,4 +51,31 @@ void isSetCookiePath() {
     var res4 = Utils.isSetCookiePath(uri4);
     assertThat(res4, equalTo(true));
   }
+
+  @Test
+  void isValidOrigin_test() {
```

Comment:
I think a few more test cases would be best, especially if the regex matching is left in.